data/json.go

```diff
@@ -152,6 +154,15 @@ type jsonConnector struct {
 
 func init() {
 	register("json", func(connStr string, l *common.Logger) (DataConnector, error) {
+
+		if _, err := os.Stat(filepath.Dir(connStr)); os.IsNotExist(err) {
```

Comment:
I think these changes are better located in the `newJsonConnector` function.

Reply:
The JSON file is already created if it doesn't exist.  And as you've mentioned, an error is thrown when an empty JSON file is found.